internal/validations/clusterlogforwarder/validate_filters.go

```diff
@@ -0,0 +1,51 @@
+package clusterlogforwarder
+
+import (
+	"regexp"
+
+	loggingv1 "github.com/openshift/cluster-logging-operator/apis/logging/v1"
+	"github.com/openshift/cluster-logging-operator/internal/validations/clusterlogforwarder/conditions"
+	"github.com/openshift/cluster-logging-operator/internal/validations/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// TODO: THIS NEEDS TO CHANGE!
+var pathExpRegex = regexp.MustCompile("^.[a-zA-Z_-][a-zA-Z0-9_-]*(.[a-zA-Z_-][a-zA-Z0-9_-]*|/[a-zA-Z_-][a-zA-Z0-9_-]*)*$")
+
+// ValidateFilters validates the defined filters.
+func ValidateFilters(clf loggingv1.ClusterLogForwarder, k8sClient client.Client, extras map[string]bool) (error, *loggingv1.ClusterLogForwarderStatus) {
+	if len(clf.Spec.Filters) == 0 {
+		return nil, nil
+	}
+
+	clf.Status.Filters = loggingv1.NamedConditions{}
+	var err error
+	for _, filterSpec := range clf.Spec.Filters {
+		// Validate the drop filter
+		if filterSpec.Type == loggingv1.FilterDrop {
+			for _, dropFilter := range *filterSpec.DropFilterSpec {
+				testErrors := []string{}
+				for _, test := range dropFilter.DropTest {
+					if !pathExpRegex.MatchString(test.Field) {
+						testErrors = append(testErrors, "field must be a valid JSONPath expression")
+					}
+					if test.Matches != "" {
```

Comment:
Don't we need a validation for when you spec both Matches and NoMatches.  We only allow one

Reply:
Yes I wasn't done with the validations, just wanted to initially get a feel for how this looks. I've added all relevant validations I think.